pkg/controllers/dnspolicy/dns_helper.go

```diff
@@ -355,3 +423,26 @@ func (r *dnsHelper) deleteDNSRecordForListener(ctx context.Context, owner metav1
 func isWildCardListener(l gatewayv1beta1.Listener) bool {
 	return strings.HasPrefix(string(*l.Hostname), "*")
 }
+
+func (dh *dnsHelper) getDNSHealthCheckProbes(ctx context.Context, gateway *gatewayv1beta1.Gateway, dnsPolicy *v1alpha1.DNSPolicy) ([]*v1alpha1.DNSHealthCheckProbe, error) {
+	list := &v1alpha1.DNSHealthCheckProbeList{}
+	if err := dh.List(ctx, list, &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(commonDNSRecordLabels(client.ObjectKeyFromObject(gateway), client.ObjectKeyFromObject(dnsPolicy))),
+		Namespace:     gateway.Namespace,
```

Comment:
The namespace here is needed for the tests. the fake client wont return the probe unless there is a Namespace ListOption. @philbrookes would we expect in a running scenario that the probes would be in a different namespace to the gateway or dnspolicy? This could probably be changed to dnsPolicy.Namespace to be honest.

Reply:
changing thisto dnspolicy.namespace seems fine to me.